app/apps/server/bridges/scheduler.js

```diff
@@ -2,22 +2,42 @@ import Agenda from 'agenda';
 import { MongoInternals } from 'meteor/mongo';
 
 function _callProcessor(processor) {
-	return (job) => processor(job.attrs.data);
+	return (job) => processor(job);
 }
 
 export class AppSchedulerBridge {
 	constructor(orch) {
 		this.orch = orch;
 		this.scheduler = new Agenda({
 			mongo: MongoInternals.defaultRemoteCollectionDriver().mongo.client.db(),
-			collection: 'rocketchat_agenda_jobs',
+			defaultConcurrency: 1,
 		});
 		this.isConnected = false;
 	}
 
 	async registerProcessors(processors = [], appId) {
+		const runAfterRegister = [];
 		this.orch.debugLog(`The App ${ appId } is registering job processors`, processors);
-		processors.forEach(({ id, processor }) => this.scheduler.define(id, _callProcessor(processor)));
+		processors.forEach(({ id, processor, startupSetting }) => {
+			this.scheduler.define(id, _callProcessor(processor));
+
+			if (startupSetting) {
+				switch (startupSetting.type) {
+					case 'onetime':
```

Comment:
This is exported by the Apps-Engine as an enum, right? I would suggest using those values instead of constants

Reply:
Done!